shared/actions/gregor.js

```diff
@@ -75,31 +83,32 @@ function checkReachabilityOnConnect() {
 }
 
 function registerGregorListeners() {
-  return (dispatch: Dispatch) => {
-    RPCTypes.delegateUiCtlRegisterGregorFirehoseRpcPromise()
-      .then(response => {
-        logger.info('Registered gregor listener')
-      })
-      .catch(error => {
-        logger.warn('error in registering gregor listener: ', error)
-      })
-
-    // we get this with sessionID == 0 if we call openDialog
-    engine().setIncomingHandler('keybase.1.gregorUI.pushState', ({state, reason}, response) => {
-      dispatch(GregorGen.createPushState({state, reason}))
-      response && response.result()
+  RPCTypes.delegateUiCtlRegisterGregorFirehoseRpcPromise()
+    .then(response => {
+      logger.info('Registered gregor listener')
+    })
+    .catch(error => {
+      logger.warn('error in registering gregor listener: ', error)
     })
 
-    engine().setIncomingHandler('keybase.1.gregorUI.pushOutOfBandMessages', ({oobm}, response) => {
-      if (oobm && oobm.length) {
-        const filteredOOBM = oobm.filter(oobm => !!oobm)
-        if (filteredOOBM.length) {
-          dispatch(GregorGen.createPushOOBM({messages: filteredOOBM}))
-        }
+  // we get this with sessionID == 0 if we call openDialog
+  engine().setIncomingActionCreators('keybase.1.gregorUI.pushState', ({reason, state}, response) => {
+    const actions = [GregorGen.createPushState({reason, state})]
+    response && response.result()
+    return actions
+  })
+
+  engine().setIncomingActionCreators('keybase.1.gregorUI.pushOutOfBandMessages', ({oobm}, response) => {
+    const actions = []
+    if (oobm && oobm.length) {
+      const filteredOOBM = oobm.filter(oobm => !!oobm)
+      if (filteredOOBM.length) {
+        actions.push(GregorGen.createPushOOBM({messages: filteredOOBM}))
```

Comment:
i think this can be triggered by favoriting and unfavoriting in kbfs

Reply:
I only see options for “Chat” and “Open”.

Reply:
I think these options have changed with the release of KBFS in the app.

Reply:
these still fire (correctly). if you go into kbfs and rmdir a folder you'll see you get a filteredOOBM fire